extra/src/main/java/aima/extra/search/pqueue/AbstractQueueSearchForActions.java

```diff
@@ -129,10 +138,10 @@ public boolean containsState(Queue<Node<A, S>> frontier, S state) {
 		return frontier.contains(state);
 	}
 
-	public boolean removedNodeFromFrontierWithSameStateAndLowerPriority(Node<A, S> child, Queue<Node<A, S>> frontier) {
-		// NOTE: Not very efficient (i.e. linear in the size of the frontier)
-		// NOTE: by Java's PriorityQueue convention, nodes that compare lower
-		// (i.e. cost) have a higher priority.
-		return frontier.removeIf(n -> child.state().equals(n.state()) && getNodeComparator().compare(child, n) < 0);
-	}
+    public boolean removedNodeFromFrontierWithSameStateAndLowerPriority(Node<A, S> child, PriorityQueue<Node<A, S>> pqFrontier) {
+        // NOTE: Not very efficient (i.e. linear in the size of the frontier)
+        // NOTE: by Java's PriorityQueue convention, nodes that compare lower
+        // (i.e. cost) have a higher priority.
+        return pqFrontier.removeIf(n -> child.state().equals(n.state()));

```

Comment:
This is incorrect. The method no longer does what it claims to do and it changes the semantics of the search algorithms that are dependent on this method to behave as it did before.